### PR TITLE
[AAP-8337] Make inventory optional

### PR DESCRIPTION
**Pull request by mkanoor**
https://issues.redhat.com/browse/AAP-8337

Since ansible-rulebook can be used with Automation Controller we don't need to make inventory mandatory. This is needed for the eda-server to launch the ansible-rulebook and execute a job on the Automation Controller.

Check if the rulebook contains any actions with run_module or run_playbook and raise an exception
```
ansible_rulebook.exception.InventoryNeededException: Rule r1 has an action run_playbook which needs inventory to be defined
```

### CODE REVIEW

**Comment by Alex-Izquierdo** [comment]
I think this is not enough, according to the ticket and discussions related, we should take care if the rulebook contains run_playbook or run_module actions

**Review by benthomasson — DISMISSED** [review]

**Comment by mkanoor** [comment]
@Alex-Izquierdo @ttuffin Please review

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+def create_condition(create_conditions, **kwargs):
+    def _condition(**kwargs):
+        when = kwargs.pop("when", "all")
+        value = kwargs.pop("value", [create_conditions()])
+        return Condition(when, value)
+
+    return _condition
+
+
+@pytest.fixture
```
Comment (bzwei) [inline tests/conftest.py:32]: @mkanoor What is the difference between `@pytest.fixture()` and `@pytest.fixture`?

**Review by Alex-Izquierdo — DISMISSED** [review]

File: tests/conftest.py (hunk 1 above)
Comment (mkanoor) [reply tests/conftest.py:32]: @bzwei They are inter changeable and uses the default scope

Comment (mkanoor) [reply tests/conftest.py:32]: @bzwei Nice catch, I have made them all consistent now.

**Review by bzwei — APPROVED** [review]

**Review by Alex-Izquierdo — APPROVED** [review]

**Review by Alex-Izquierdo — APPROVED** [review]

**Review by bzwei — APPROVED** [review]

**Pull request merged**